plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergTableWithCustomLocation.java

```diff
@@ -52,125 +43,91 @@
-        HdfsConfiguration hdfsConfiguration = new HiveHdfsConfiguration(new HdfsConfigurationInitializer(hdfsConfig), ImmutableSet.of());
-        HdfsEnvironment hdfsEnvironment = new HdfsEnvironment(hdfsConfiguration, hdfsConfig, new NoHdfsAuthentication());
-
-        metastore = new FileHiveMetastore(
-                new NodeVersion("test_version"),
-                hdfsEnvironment,
-                new MetastoreConfig(),
-                new FileHiveMetastoreConfig()
-                        .setCatalogDirectory(baseDir.toURI().toString())
-                        .setMetastoreUser("test"));
-
-        TestingIcebergPlugin icebergPlugin = new TestingIcebergPlugin(metastore, false);
-
-        queryRunner.installPlugin(icebergPlugin);
-        queryRunner.createCatalog(CATALOG, "iceberg", ImmutableMap.of("iceberg.unique-table-location", "true"));
-        queryRunner.installPlugin(new TestingHivePlugin(metastore));
-        queryRunner.createCatalog("hive", "hive", ImmutableMap.of("hive.security", "sql-standard"));
-        return queryRunner;
-    }
-
-    @BeforeClass
-    public void setUp()
-    {
-        assertQuerySucceeds(format("CREATE SCHEMA %s.%s", CATALOG, SCHEMA));
-    }
-
-    @AfterMethod
-    public void clean()
-    {
-        query(format("DROP TABLE IF EXISTS %s.%s.%s", CATALOG, SCHEMA, TABLE));
-        query(format("DROP TABLE IF EXISTS %s.%s.%s", CATALOG, SCHEMA, TABLE_RENAMED));
+        File tempDir = Files.createTempDirectory("test_iceberg_v2").toFile();
+        metastoreDir = new File(tempDir, "iceberg_data");
+        metastore = createTestingFileHiveMetastore(metastoreDir);
+
+        return createIcebergQueryRunner(
+                ImmutableMap.of(),
+                ORC,
+                ImmutableList.of(NATION),
+                Optional.of(metastoreDir),
+                Optional.of(ImmutableMap.of("iceberg.unique-table-location", "true")));
     }
 
     @AfterClass(alwaysRun = true)
-    public void tearDown()
+    public void tearDown() throws IOException
     {
-        query(format("DROP SCHEMA %s.%s", CATALOG, SCHEMA));
+        deleteRecursively(metastoreDir.getParentFile().toPath(), ALLOW_INSECURE);
     }
 
     @Test
     public void testTableHasUuidSuffixInLocation()
     {
-        assertQuerySucceeds(format("CREATE TABLE %s.%s.%s as select 1 as val", CATALOG, SCHEMA, TABLE));
-        Optional<Table> table = metastore.getTable(null, SCHEMA, TABLE);
+        assertQuerySucceeds("CREATE TABLE table_with_uuid as select 1 as val");
+        Optional<Table> table = metastore.getTable(null, "tpch", "table_with_uuid");
         assertTrue(table.isPresent(), "Table should exists");
         String location = table.get().getStorage().getLocation();
-        assertTrue(location.matches(format(".*%s-[0-9a-f]{32}", TABLE)), "Table location should have UUID suffix");
+        assertTrue(location.matches(format(".*%s-[0-9a-f]{32}", "table_with_uuid")), "Table location should have UUID suffix");
```

Comment:
Do not use `assertTrue` with complex expressions. In case of failure, it produces unhelpful message.

Use 
```
[org.assertj.core.api.Assertions.] assertThat(location).matches(....);
```


instead

Reply:
done